nucliadb/nucliadb/search/search/query.py

```diff
@@ -67,7 +67,12 @@ async def global_query_to_pb(
 
     # We need to ask for all and cut later
     request.page_number = 0
-    request.result_per_page = page_number * page_size + page_size
+    if sort is not None:
+        # As the index can't sort, we have to do it when merging. To
+        # have consistent results, we must limit them
+        request.result_per_page = sort.limit
```

Comment:
what happens if page_size provided by the user is >= sort.limit?

Reply:
They won't have any result. The idea here is that the user is requesting the `sort.limit` most relevant results and then sorting them by some criteria. If you want more, you need to request more.